### PR TITLE
Add PHP8.2 to CI matrix

### PR DESCRIPTION
**Pull request by ChristophWurst**
Proof of concept for https://github.com/nextcloud/groupware/issues/40.

### CODE REVIEW

**Comment by ChristophWurst** [comment]
I didn't expect this to "just work" but let's go

**Comment by kesselb** [comment]
Pushed a commit to patch the version check before nextcloud setup.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -128,6 +136,9 @@ jobs:
                 coverage: xdebug
           - name: Checkout Nextcloud
             run: git clone https://github.com/nextcloud/server.git --recursive --depth 1 -b ${{ matrix.nextcloud-versions }} nextcloud
+          - name: Patch version check for nightly PHP
+                  if: ${{ matrix.php-versions == '8.2' }}
```
Comment (kesselb) [inline .github/workflows/test.yml:140]: That looks off ;)

Comment (ChristophWurst) [reply .github/workflows/test.yml]: While fixing it squash the commits right away. I think CI should pass now

**Comment by ChristophWurst** [comment]
> App "Mail" cannot be installed because the following dependencies are not fulfilled: PHP with a version lower than 8.1 is required.

we have to patch info.xml

**Comment by kesselb** [comment]
> App "Mail" cannot be installed because the following dependencies are not fulfilled: PHP with a version lower than 8.1 is required.

Added another commit to patch max-version in appinfo.

**Pull request merged**